include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -168,6 +168,12 @@ ERROR(error_formatting_multiple_file_ranges,none,
 ERROR(error_formatting_invalid_range,none,
   "file range is invalid", ())
 
+ERROR(invalid_conditional_compilation_flag,none,
+      "invalid conditional compilation flag in '-D%0'", (StringRef))
+
+ERROR(cannot_assign_value_to_conditional_compilation_flag,none,
+      "conditional compilation flags cannot get a specific value assigned in '-D%0'", (StringRef))
```

Comment:
Nitpick: the grammar sounds a little funny to me. Maybe just "conditional compilation flags do not have values in Swift; they are either present or absent". Though I suppose the value the user provided has to be worked in there somewhere, since otherwise they won't know what it's referring to.